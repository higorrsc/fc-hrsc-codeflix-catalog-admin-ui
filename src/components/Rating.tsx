import { Box, Typography } from '@mui/material';

type Props = {
  rating: 'L' | '10' | '12' | '14' | '16' | '18';
};

const backgroundColors = {
  L: '#39B549',
  '10': '#20A3D4',
  '12': '#E79738',
  '14': '#E35E00',
  '16': '#D00003',
  '18': '#000000',
};

export const Rating = ({ rating }: Props) => {
  return (
    <Box
      sx={{
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: backgroundColors[rating],
        borderRadius: '4px',
        height: 48,
        width: 48,
      }}
    >
      <Typography>{rating}</Typography>
    </Box>
  );
};
