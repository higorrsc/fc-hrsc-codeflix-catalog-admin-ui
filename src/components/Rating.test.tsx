import { render, screen } from '@testing-library/react';
import { Rating } from './Rating';

const backgroundColors = {
  L: 'rgb(57, 181, 73)', // #39B549
  '10': 'rgb(32, 163, 212)', // #20A3D4
  '12': 'rgb(231, 151, 56)', // #E79738
  '14': 'rgb(227, 94, 0)', // #E35E00
  '16': 'rgb(208, 0, 3)', // #D00003
  '18': 'rgb(0, 0, 0)', // #000000
};

type RatingValue = keyof typeof backgroundColors;

describe('Test Rating Component', () => {
  it('should render correctly for rating "L" and match snapshot', () => {
    const { asFragment } = render(<Rating rating='L' />);
    const ratingElement = screen.getByText('L');
    expect(ratingElement).toBeInTheDocument();
    expect(ratingElement.parentElement).toHaveStyle(
      `background-color: ${backgroundColors.L}`
    );
    expect(asFragment()).toMatchSnapshot();
  });

  const ratingsToTest: RatingValue[] = ['10', '12', '14', '16', '18'];

  ratingsToTest.forEach((ratingValue) => {
    it(`should render correctly for rating "${ratingValue}"`, () => {
      render(<Rating rating={ratingValue} />);
      const ratingElement = screen.getByText(ratingValue);
      expect(ratingElement).toBeInTheDocument();
      expect(ratingElement.parentElement).toHaveStyle(
        `background-color: ${backgroundColors[ratingValue]}`
      );
    });
  });

  // Example of how you might test for specific styles if needed beyond background color
  it('should have correct common styling', () => {
    render(<Rating rating='L' />);
    const boxElement = screen.getByText('L').parentElement;

    expect(boxElement).toHaveStyle({
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      borderRadius: '4px',
      height: '48px',
      width: '48px',
    });
  });

  it('should display the rating text inside the box', () => {
    render(<Rating rating='14' />);
    const typographyElement = screen.getByText('14');
    expect(typographyElement).toBeInTheDocument();
    expect(typographyElement.tagName).toBe('P'); // MUI Typography renders a <p> by default for body2
  });
});
