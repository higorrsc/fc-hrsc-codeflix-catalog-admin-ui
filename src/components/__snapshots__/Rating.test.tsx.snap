// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Rating Component should render correctly for rating "L" and match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-pz392n"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
    >
      L
    </p>
  </div>
</DocumentFragment>
`;
